### ISSUE TITLE
Add Iterant.dropLast

### ISSUE THREAD

**Issue by alexandru** [monix/monix#495]
We need an `Iterant` op that drops the last `n: Int` elements from the stream.

The method signature should be something like this:

```scala
sealed abstract class Iterant[F[_], A] {
  // ...
  def dropLast(n: Int)(implicit F: Sync[F]): Iterant[F, A]
}
```

For inspiration check out [DropLastOperator](https://github.com/monix/monix/blob/v3.0.0-M3/monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/DropLastOperator.scala), the implementation for `Observable`.

**Comment by martin-g** [comment]
Sorry for the noise but what is the idea behind re-using `F` in `implicit F: Sync[F]` ? 
Why the method parameter (`implicit F`) is named the same as the type parameter (`F[_]`) ?

**Comment by alexandru** [comment]
@martin-g it's just a convention that I've seen in other Scala codebases and that I also adopted, related to typeclasses. I first saw it in [iteratee.io](https://github.com/travisbrown/iteratee) and also in [FS2](https://github.com/functional-streams-for-scala/fs2/).

It's very subjective, but I think it's nice to then see in the implementation things like ...

```scala
F.map(this)(f)
```

Which I sometimes prefer instead of using `cats.syntax`.
